zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/ZoneService.java

```diff
@@ -29,6 +113,268 @@ public ZoneResponse getZoneDetails(String zoneName) {
+        try {
+            created = zoneProvisioning.create(identityZone);
+        } catch (Exception e) {
+            logger.error("Unable to create identity zone for subdomain : " + subdomain + " Exception is : " + e.getMessage());
+            throw new ZoneServiceException("Unexpected exception while create identity zone for " +
+                                           "subdomain : ." + subdomain + " Exception is : " + e.getMessage());
+        }
+
+        if(created != null){
+            try {
+                IdentityProvider defaultIdp = new IdentityProvider();
+                defaultIdp.setName(OriginKeys.UAA);
+                defaultIdp.setType(OriginKeys.UAA);
+                defaultIdp.setOriginKey(OriginKeys.UAA);
+                defaultIdp.setIdentityZoneId(created.getId());
+                UaaIdentityProviderDefinition idpDefinition = new UaaIdentityProviderDefinition();
+                idpDefinition.setPasswordPolicy(null);
+                defaultIdp.setConfig(idpDefinition);
+                idpProvisioning.create(defaultIdp, created.getId());
+            } catch (Exception e) {
+                logger.error("Unable to create identity provider for subdomain : " + subdomain + " Exception is : " + e.getMessage());
+                throw new ZoneServiceException("Unexpected exception while create identity provider for " +
+                                               "subdomain : ." + subdomain + " Exception is : " + e.getMessage());
+            }
+
+            createUserGroups(created);
+
+            String zoneId = IdentityZoneHolder.get().getId();
+            String authorities = zoneAuthorities + ",zones." + zoneId + ".admin";
+
+            SimpleGrantedAuthority authority = new SimpleGrantedAuthority("zones." + id + ".admin");
+            List<SimpleGrantedAuthority> tempAuthorityList = Arrays.asList(new SimpleGrantedAuthority[] { authority });
+            List<SimpleGrantedAuthority> emptyAuthorityList = Arrays.asList(new SimpleGrantedAuthority[] {});
+
+            try {
+                updateDefaultAdminAuthorities(tempAuthorityList, emptyAuthorityList, zoneId);
+                createClient(created.getId(), authorities, clientId, adminClientSecret, grantTypes, resourceIds,
+                             scopes);
+            } catch (Exception e) {
+                logger.error("Unable to create client for subdomain : " + subdomain + " Zone Id : "+zoneId+" Exception is :" + e.getMessage());
+                throw new ZoneServiceException("Unexpected exception while create client for " +
+                                               "subdomain : " + subdomain + " Zone Id : "+zoneId+" Exception is :" + e.getMessage());
+            } finally{
+                updateDefaultAdminAuthorities(emptyAuthorityList, tempAuthorityList, zoneId);
+            }
+        }
+    }
+
+    public void updateDefaultAdminAuthorities(final List<SimpleGrantedAuthority> addAuthorities,
```

Comment:
This is done to elevate authorities of the base zone admin client so it could be used to administer newly created zone, like add an admin client.  Since here you are calling clientDetailsService.create() directly without using REST API and without token, I suspect it's not needed.

Reply:
Yes, its not required, i have removed it accordingly. Hence these are there for broker set the authorities as rest url calls. thanks!